src/codemagic/tools/app_store_connect.py

```diff
@@ -492,25 +492,53 @@ def list_devices(
         "register-device",
         BundleIdArgument.PLATFORM,
         DeviceArgument.DEVICE_NAME,
-        DeviceArgument.DEVICE_UDID,
+        DeviceArgument.DEVICE_UDIDS,
+        DeviceArgument.IGNORE_REGISTRATION_ERRORS,
     )
     def register_device(
         self,
         platform: BundleIdPlatform,
         device_name: str,
-        device_udid: str,
+        device_udids_argument: Union[str, Types.DeviceUdidsArgument, Sequence[Union[str, Types.DeviceUdidsArgument]]],
+        ignore_registration_errors: bool = False,
         should_print: bool = True,
-    ) -> Device:
+    ) -> List[Device]:
         """
-        Register a new device for app development
+        Register new Devices for app development
         """
-        return self._create_resource(
-            self.api_client.devices,
-            should_print,
-            udid=device_udid,
-            name=device_name,
-            platform=platform,
-        )
+        if isinstance(device_udids_argument, str) or isinstance(device_udids_argument, Types.DeviceUdidsArgument):
+            device_udids_argument = [device_udids_argument]
+
+        device_udids = []
+        for device_udids_argument in device_udids_argument:
+            if isinstance(device_udids_argument, Types.DeviceUdidsArgument):
+                device_udids.extend(device_udids_argument.value)
+            else:
+                device_udids.append(device_udids_argument)
+        if not device_udids:
+            DeviceArgument.DEVICE_UDIDS.raise_argument_error("At least one device UDID is required")
+
+        registered_devices = []
+        for i, device_udids_argument in enumerate(device_udids):
+            if should_print:
+                self.echo("") if i != 0 else None
```

Comment:
I find that it would be easier to remove the enumeration and do this echoing after the try-except-else block. Empty line after the last registration is fine.